dymos/run_problem.py

```diff
@@ -109,24 +112,33 @@ def run_problem(problem, refine=False, refine_iteration_limit=10, run_driver=Tru
 
         phases = find_phases(problem.model)
 
-        ref = PHAdaptive(phases)
+        ref = refinement_methods[refine_method](phases)
         with open(out_file, 'w+') as f:
 
             for i in range(refine_iteration_limit):
                 refine_results = check_error(phases)
 
-                ref.refine(refine_results)
-
-                for stream in f, sys.stdout:
-                    ref.write_iteration(stream, i, phases, refine_results)
-
                 refined_phases = [phase_path for phase_path in refine_results if
                                   phases[phase_path].refine_options['refine'] and
                                   np.any(refine_results[phase_path]['need_refinement'])]
 
                 if not refined_phases:
                     break
 
+                if refine_method == 'hp':
+                    for phase_path in refine_results:
+                        ref.error[phase_path] = refine_results[phase_path]['max_rel_error']
+                    if i == 0:
+                        ref.refine_first_iter(refine_results)
+                    else:
+                        ref.refine(refine_results)
+                    ref.previous_error = ref.error.copy()
+                else:
+                    ref.refine(refine_results)
+
+                for stream in f, sys.stdout:
+                    ref.write_iteration(stream, i, phases, refine_results)
```

Comment:
We should probably break the `write_iteration` method into two parts.  As it is now, we don't get any indication from the grid refinement about the error on the last refinement pass, just "refinement successfully completed".

After `check_error` we should write out the portions of the refine_results that deal with the current grid and the associated error.

If refinement is necessary, `refine` should then be called and information about the new grid printed out.